cardano-cli/src/Cardano/CLI/Types.hs

```diff
@@ -220,17 +227,16 @@ instance Crypto.Crypto crypto =>  ToJSON (Params crypto) where
     , "retiring" .= r
     ]
 
-newtype SigningKeyFile = SigningKeyFile
-  { unSigningKeyFile :: FilePath }
-  deriving stock (Eq, Ord)
-  deriving newtype (IsString, Show)
+data OfSigningKey
+
+type SigningKeyFile = File OfSigningKey
```

Comment:
Instead of being a `newtype` with its own set of instances we will use a type alias to `File` with the file type `content` already supplied.  This means it's possible to use file operation functions like `writeByteStringFile` directly because they accept any `content`.